packages/asset/contracts/LazyVault.sol

```diff
@@ -0,0 +1,131 @@
+    function distribute(
+        uint8[] calldata tiers,
+        uint256[] calldata amounts,
+        address[] calldata creators
+    ) external onlyRole(DISTRIBUTOR_ROLE) {
+        require(tiers.length == amounts.length, "LazyVault: 1-Array mismatch");
+        require(tiers.length == creators.length, "LazyVault: 2-Array mismatch");
+
+        for (uint256 i = 0; i < tiers.length; i++) {
+            require(tiers[i] > 0 && tiers[i] < tierValues.length, "LazyVault: Invalid tier");
+            uint256 totalValue = amounts[i] * tierValues[tiers[i]];
+            uint256 creatorShare = totalValue;
+            uint256[] memory splitValues = new uint256[](splitRecipients.length);
+            address[] memory splitRecipientsAddresses = new address[](splitRecipients.length);
+            for (uint256 j = 0; j < splitRecipients.length; j++) {
+                SplitRecipient memory recipient = splitRecipients[j];
+                splitRecipientsAddresses[j] = recipient.recipient;
+                uint256 splitValue = (totalValue * recipient.split) / 10000;
+                splitValues[j] = splitValue;
+                IERC20(vaultToken).transfer(recipient.recipient, splitValue);
+                creatorShare -= splitValue;
+            }
+
+            IERC20(vaultToken).transfer(creators[i], creatorShare);
+
+            emit Distributed(
+                tiers[i],
+                amounts[i],
+                tierValues[tiers[i]],
+                creators[i],
+                splitValues,
+                splitRecipientsAddresses,
+                creatorShare
+            );
+        }
+    }
+
+    /// @notice Withdraws tokens from the vault
+    /// @dev Only the manager role can withdraw tokens
+    /// @param amount The amount to withdraw
+    function withdraw(uint256 amount) external onlyRole(MANAGER_ROLE) {
+        IERC20(vaultToken).transfer(_msgSender(), amount);
+        emit VaultWithdraw(_msgSender(), amount);
+    }
+
+    /// @notice Changes the split recipients and their split amount
+    /// @dev Only the manager role can change the split recipients
+    /// @param recipients The new split recipients
+    function changeSplitRecipients(SplitRecipient[] calldata recipients) external onlyRole(MANAGER_ROLE) {
+        _setSplitRecipients(recipients);
+    }
+
+    /// @dev Wipes the splitRecipients array and sets the new recipients
+    /// @param recipients The new split recipients
+    function _setSplitRecipients(SplitRecipient[] memory recipients) internal {
+        delete splitRecipients;
+        for (uint256 i = 0; i < recipients.length; i++) {
+            SplitRecipient memory recipient = recipients[i];
+            require(recipient.split <= 10000, "Split must be <= 10000");
+            require(recipient.recipient != address(0), "Recipient cannot be 0x0");
+            splitRecipients.push(recipient);
+        }
+        emit SplitRecipientsChanged(splitRecipients);
+    }
```

Comment:
should the total splits equal to 10 000 ?

Reply:
No, this is just for TSB recipients.